Go/Go.sublime-syntax

```diff
@@ -186,25 +190,67 @@ contexts:
       scope: variable.other.go
 
   # https://golang.org/ref/spec#Predeclared_identifiers
-  match-predefined-identifiers:
-    # Note: these are NOT keywords, but rather predeclared identifiers that can
-    # be redefined. We treat them as regular identifiers in variable
-    # declarations, types, function names, etc. This rule should be used in
-    # places that are "left over". Detecting redefinition is beyond the scope of
-    # this syntax engine.
+  #
+  # In Go, the predeclared constants are not keywords, and can be redefined. In
+  # many places such as variable declarations, types, function names, etc, we
+  # allow them to be scoped the same way as other identifiers. This "constant"
+  # rule should be used in places that are "left over". Detecting redefinition
+  # would be ideal, but is beyond the scope of this syntax engine; we simply
+  # expect it to be very rare.
+  match-predeclared-constants:
     - match: \b(?:true|false|nil)\b
       scope: constant.language.go
-    - include: match-new
-    - include: match-make
 
-  # Note: calls and casts are syntactically identical. Detecting casts and
-  # scoping them as types is beyond the capabilities of this syntax engine.
+  # Note: in Go, calls and casts are syntactically identical. Detecting casts
```

Comment:
Note that Go does not have casts. Too late for this change though.

Reply:
Go's "type conversions" are a superset of C casts. I don't understand why people insist on Go "not having casts". The term actually means different things in different languages. For example, Java's casts are a superset of Go's "type conversions". I wish Go designers just used the shorter term and chilled out.

Reply:
I guess there's no reason not to update the comments and context names. The syntax does claim to reflect the spec's lexicon.

Reply:
Also, the term is misleading to begin with, IMO. Subjectively, "type conversions" implies that only the type is converted, while the value remains unchanged. This is untrue for numeric conversions (which insert specially generated assembly and/or built-in runtime functions), string and byte conversions (which call built-in runtime functions that allocate and memcpy), and conversions involving interfaces.